src/lib/encoder.ts

```diff
@@ -74,23 +74,58 @@ const encodeDataSourceWithHash = (
   const verificationMethod = getVerificationMethod(
     verification?.method || UNKNOWN_VERIFICATION_METHOD,
   );
-  return (
-    (verificationMethod
-      ? keccak256(verificationMethod.name).slice(0, 10)
-      : padLeft(0, 8)) +
-    stripHexPrefix(verification ? verification.data : padLeft(0, 64)) +
-    stripHexPrefix(utf8ToHex(dataSource))
-  );
+  return [
+    '0x0000',
+    stripHexPrefix(
+      verificationMethod
+        ? padLeft(keccak256(verificationMethod.name).slice(0, 10), 8)
+        : '00000000',
+    ),
+    stripHexPrefix(
+      verification?.data
+        ? padLeft(verification.data.slice(2).length / 2, 4)
+        : '0000',
+    ),
+    stripHexPrefix(
+      verification?.data ? stripHexPrefix(verification?.data) : '',
+    ),
+    stripHexPrefix(utf8ToHex(dataSource)),
+  ].join('');
 };
 
 const decodeDataSourceWithHash = (value: string): URLDataWithHash => {
+  if (value.slice(0, 6) === '0x0000') {
+    /*
+      0        1         2         3         4         5         6         7         8
+      12345678901234567890123456789012345678901234567890123456789012345678901234567890
+      0x0000 code
+            6f357c6a hash fn [6]
+                    0020 data len [14]
+                        820464ddfac1be...[18 + data len]
+                                                       [18 + data len]...696670733a2f2...[...rest]
+    */
+    const verificationMethodSig = `0x${value.slice(6, 14)}`;
```

Comment:
What is `Sig` abbreviation for ?

Reply:
Signature. Normally the function signature is the method name ( param types... ) and then the hash of it is sometimes called sig or hash (at least in some of the public type sets)